diff --git a/fs/exec.c b/fs/exec.c
index d834acacc69e..19ad3595ebe5 100644
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1261,18 +1261,21 @@ void __set_task_comm(struct task_struct *tsk, const char *buf, bool exec)
 	task_lock(tsk);
 
 #ifdef CONFIG_PERF_HUMANTASK
-	if (!strcmp(parent->comm, "system_server")) {
-		if (!strcmp(buf, "InputDispatcher") ||
-		    !strcmp(buf, "InputReader")) {
-			tsk->human_task = MAX_LEVER + 1;
-		} else if (tmpbuf) {
-			memset(tmpbuf, 0, 128);
-			sprintf(tmpbuf, "Binder:%d_%X", tsk->tgid, 1);
+    if (parent) {
+	    if (!strcmp(parent->comm, "system_server")) {
+		    if (!strcmp(buf, "InputDispatcher") ||
+		        !strcmp(buf, "InputReader")) {
+			    tsk->human_task = MAX_LEVER + 1;
+		    } else if (tmpbuf) {
+			    memset(tmpbuf, 0, 128);
+			    sprintf(tmpbuf, "Binder:%d_%X", tsk->tgid, 1);
 			// binder/ProcessState.cpp
-			if (!strcmp(tmpbuf, buf))
-				tsk->human_task = 1;
-		}
-	}
+			    if (!strcmp(tmpbuf, buf))
+				    tsk->human_task = 1;
+		    }
+	    }
+    }
+    	
 	if (tmpbuf)
 		kfree(tmpbuf);
 #endif