--- a/drivers/staging/qcacld-3.0/Kbuild
+++ b/drivers/staging/qcacld-3.0/Kbuild
@@ -3353,7 +3353,7 @@
 CLD_CHECKOUT = $(shell cd "$(WLAN_ROOT)" && \
 	git reflog | grep -vm1 "}: cherry-pick: " | grep -oE ^[0-f]+)
 CLD_IDS = $(shell cd "$(WLAN_ROOT)" && \
-	git log -50 $(CLD_CHECKOUT)~..HEAD | \
+	git log -50 $(CLD_CHECKOUT)..HEAD | \
 		sed -nE 's/^\s*Change-Id: (I[0-f]{10})[0-f]{30}\s*$$/\1/p' | \
 		paste -sd "," -)
 
@@ -3360,7 +3360,7 @@
 CMN_CHECKOUT = $(shell cd "$(WLAN_COMMON_INC)" && \
 	git reflog | grep -vm1 "}: cherry-pick: " | grep -oE ^[0-f]+)
 CMN_IDS = $(shell cd "$(WLAN_COMMON_INC)" && \
-	git log -50 $(CMN_CHECKOUT)~..HEAD | \
+	git log -50 $(CMN_CHECKOUT)..HEAD | \
 		sed -nE 's/^\s*Change-Id: (I[0-f]{10})[0-f]{30}\s*$$/\1/p' | \
 		paste -sd "," -)