diff --git a/kernel/Makefile b/kernel/Makefile
index 36a927694a8c..7f357ebb32bf 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -156,7 +156,7 @@ $(obj)/config_data.gz: arch/arm64/configs/umi_stock-defconfig FORCE
 else ifeq ($(CONFIG_MACH_XIAOMI_PIPA),y)
 $(obj)/config_data.gz: arch/arm64/configs/pipa_stock-defconfig FORCE
 else ifeq ($(CONFIG_MACH_XIAOMI_DAGU),y)
-$(obj)/config_data.gz: arch/arm64/configs/dagu_stock-defconfig FORCE
+$(obj)/config_data.gz: $(KCONFIG_CONFIG) FORCE
 else
 $(obj)/config_data.gz: $(KCONFIG_CONFIG) FORCE
 endif
